compiler/rustc_resolve/src/late.rs

```diff
@@ -197,7 +197,7 @@ impl<'a> PathSource<'a> {
     fn namespace(self) -> Namespace {
         match self {
             PathSource::Type | PathSource::Trait(_) | PathSource::Struct => TypeNS,
-            PathSource::Expr(..) | PathSource::Pat | PathSource::TupleStruct(_) => ValueNS,
+            PathSource::Expr(..) | PathSource::Pat | PathSource::TupleStruct(_, _) => ValueNS,
```

Comment:
```suggestion
            PathSource::Expr(..) | PathSource::Pat | PathSource::TupleStruct(..) => ValueNS,
```
Here and in other places.

Reply:
>and in other places.

Not addressed yet.